lib/bankaccount/checkingaccount.rb

```diff
@@ -0,0 +1,35 @@
+module BankAccount
+  class CheckingAccount < Account
+    attr_reader :checks_used
+
+    def initialize(id, initial_balance, required_opening_balance = 0) # required_opening_balance = 0 is not necessary, since this matches base class's default, but I wanted to be explicit
+      super
+      @checks_used = 0
+    end
+
+    def withdraw(amount, transaction_fee = 1, min_balance = 0) # [lunch]
+      super
+    end
+
+    def withdraw_using_check(amount) # [lunch]
+      withdraw(amount, check_fee, -10) # (amount, transaction_fee, min_balance)
+
+      # increment @checks_used *after* check for sufficient funds
+      @checks_used += 1
+      return @balance
```

Comment:
Since Ruby returns the last line of a method by default, you don't need to say `return` explicitly--you can just say `@balance`.